#### The aim of this project is to predict the energy use of lights and appliances in a low-energy building using the UCI data set collected in 2017. This dataset is provided to us by tutoring team of Computational Machine Learning, Master of AI at RMIT University. The description of the dataset was also found within the same resources provided to us.

In [2]:
# Importing the necessary libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

### Retrieving the dataset

In [5]:
""" As per the scope of this of this project, the provided dataset is enough and we do not require any other
    to conduct and express our analysis of this problem. """

df = pd.read_csv('UCI_data.csv')

#### The description of the dataset as provided with the resources of the project.

date time year-month-day hour:minute:second <br>
T1 -  Temperature in kitchen area, in Celsius <br>
RH_1 -  Humidity in kitchen area, in % <br>
T2 - Temperature in living room area, in Celsius <br> 
RH_2 - Humidity in living room area, in % <br>
T3 -  Temperature in laundry room area <br>
RH_3 -  Humidity in laundry room area, in % <br>
T4 -  Temperature in office room, in Celsius <br>
RH_4 -  Humidity in office room, in % <br>
T5 - Temperature in bathroom, in Celsius <br>
RH_5 - Humidity in bathroom, in % <br>
T6 - Temperature outside the building (north side), in Celsius <br>
RH_6 - Humidity outside the building (north side), in % <br>
T7 - Temperature in ironing room , in Celsius <br>
RH_7 - Humidity in ironing room, in % <br>
T8 - Temperature in teenager room 2, in Celsius <br>
RH_8 - Humidity in teenager room 2, in % <br>
T9 - Temperature in parents room, in Celsius <br>
RH_9 - Humidity in parents room, in % <br>
To - Temperature outside (from Chievres weather station), in Celsius <br>
Pressure - (from Chievres weather station), in mm Hg <br>
RH_out - Humidity outside (from Chievres weather station), in % <br>
Wind speed - (from Chievres weather station), in m/s <br>
Visibility - (from Chievres weather station), in km <br>
Tdewpoint - (from Chievres weather station), °C <br>
rv1 - Random variable 1, nondimensional <br>
rv2 - Random variable 2, nondimensional <br>
<br>
------------------------------------------------------------------------------ <br>
TARGET_Energy - energy use of Appliances and light fixtures in the house in Wh

### Data Preparation

In [18]:
# Printing the name and the number of columns

print("Number of columns:- {} columns".format(len(df.columns)))
print("Column Names:- ",", ".join(list(df.columns)))

Number of columns:- 28 columns
Column Names:-  date, T1, RH_1, T2, RH_2, T3, RH_3, T4, RH_4, T5, RH_5, T6, RH_6, T7, RH_7, T8, RH_8, T9, RH_9, T_out, Press_mm_hg, RH_out, Windspeed, Visibility, Tdewpoint, rv1, rv2, TARGET_energy


In [26]:
""" As there are a lot of columns and we may require to print all of them, we will increase the limit
    of the number of columns that can be printed at one time """

pd.set_option("display.max_columns", None)

In [19]:
# Printing some rows to get a basic idea of the data

df.head()

date         T1       RH_1         T2       RH_2         T3  \
0  2016-04-19 20:30:00  22.200000  39.500000  20.566667  37.656667  22.230000   
1  2016-03-05 04:40:00  20.356667  37.126667  17.566667  40.230000  20.890000   
2  2016-03-14 12:40:00  20.926667  38.790000  21.100000  35.526667  21.600000   
3  2016-01-22 15:30:00  18.290000  38.900000  17.290000  39.260000  18.390000   
4  2016-02-10 00:40:00  22.290000  42.333333  21.600000  40.433333  22.666667   

        RH_3         T4       RH_4         T5  ...   RH_9  T_out  Press_mm_hg  \
0  37.030000  22.318571  36.610000  20.633333  ...  33.90   9.70   766.100000   
1  37.663333  18.700000  36.260000  18.463333  ...  41.09   0.30   740.333333   
2  36.290000  21.000000  34.826667  18.100000  ...  38.76   4.40   768.466667   
3  39.326667  16.100000  38.790000  16.100000  ...  39.20   3.35   760.600000   
4  43.363333  19.100000  40.900000  19.290000  ...  43.73   3.20   738.900000   

   RH_out  Windspeed  Visibility  Tdewpoint        rv1        rv2  \
0    65.5   3.500000   40.000000   3.350000  24.061869  24.061869   
1    99.0   1.000000   41.333333   0.100000   4.622052   4.622052   
2    72.0   6.000000   22.666667  -0.266667   5.635898   5.635898   
3    82.0   5.500000   41.000000   0.500000  49.216445  49.216445   
4    88.0   7.333333   56.000000   1.400000  47.617579  47.617579   

   TARGET_energy  
0             60  
1             50  
2             80  
3             40  
4             60  

[5 rows x 28 columns]

From the data printed above, we may say that the dataset represents a time-series with the column 'date' holding the date and time. Most of the columns seem to hold float values and the target columns represents values in the form of integers with no decimal values. However, the range of some columns appear to be different in accordance with the others and hence scaling may be required.

In [21]:
# Checking the size of the dataset.

df_len = len(df)
df_len

19735

Therefore, the dataset is of shape (19735,28).

In [22]:
# Extracting out some basic information from the dataset.

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19735 entries, 0 to 19734
Data columns (total 28 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   date           19735 non-null  object 
 1   T1             19735 non-null  float64
 2   RH_1           19735 non-null  float64
 3   T2             19735 non-null  float64
 4   RH_2           19735 non-null  float64
 5   T3             19735 non-null  float64
 6   RH_3           19735 non-null  float64
 7   T4             19735 non-null  float64
 8   RH_4           19735 non-null  float64
 9   T5             19735 non-null  float64
 10  RH_5           19735 non-null  float64
 11  T6             19735 non-null  float64
 12  RH_6           19735 non-null  float64
 13  T7             19735 non-null  float64
 14  RH_7           19735 non-null  float64
 15  T8             19735 non-null  float64
 16  RH_8           19735 non-null  float64
 17  T9             19735 non-null  float64
 18  RH_9  

From the information above, we can confirm that all the columns except the date column and the target variable are of type 'float64'. The date column belongs to the object class specifically and it must be in the perfect format to be casted into the 'datetime' object. The target variable is an integer type. The non-null count of each column is equal to the number of rows present in the dataset which concludes that there are <b>no missing values</b>.

In [28]:
# Describing the dataset in statistical terms.

df.describe()

T1          RH_1            T2          RH_2            T3  \
count  19735.000000  19735.000000  19735.000000  19735.000000  19735.000000   
mean      21.686571     40.259739     20.341219     40.420420     22.267611   
std        1.606066      3.979299      2.192974      4.069813      2.006111   
min       16.790000     27.023333     16.100000     20.463333     17.200000   
25%       20.760000     37.333333     18.790000     37.900000     20.790000   
50%       21.600000     39.656667     20.000000     40.500000     22.100000   
75%       22.600000     43.066667     21.500000     43.260000     23.290000   
max       26.260000     63.360000     29.856667     56.026667     29.236000   

               RH_3            T4          RH_4            T5          RH_5  \
count  19735.000000  19735.000000  19735.000000  19735.000000  19735.000000   
mean      39.242500     20.855335     39.026904     19.592106     50.949283   
std        3.254576      2.042884      4.341321      1.844623      9.022034   
min       28.766667     15.100000     27.660000     15.330000     29.815000   
25%       36.900000     19.530000     35.530000     18.277500     45.400000   
50%       38.530000     20.666667     38.400000     19.390000     49.090000   
75%       41.760000     22.100000     42.156667     20.619643     53.663333   
max       50.163333     26.200000     51.090000     25.795000     96.321667   

                 T6          RH_6            T7          RH_7            T8  \
count  19735.000000  19735.000000  19735.000000  19735.000000  19735.000000   
mean       7.910939     54.609083     20.267106     35.388200     22.029107   
std        6.090347     31.149806      2.109993      5.114208      1.956162   
min       -6.065000      1.000000     15.390000     23.200000     16.306667   
25%        3.626667     30.025000     18.700000     31.500000     20.790000   
50%        7.300000     55.290000     20.033333     34.863333     22.100000   
75%       11.256000     83.226667     21.600000     39.000000     23.390000   
max       28.290000     99.900000     26.000000     51.400000     27.230000   

               RH_8            T9          RH_9         T_out   Press_mm_hg  \
count  19735.000000  19735.000000  19735.000000  19735.000000  19735.000000   
mean      42.936165     19.485828     41.552401      7.411665    755.522602   
std        5.224361      2.014712      4.151497      5.317409      7.399441   
min       29.600000     14.890000     29.166667     -5.000000    729.300000   
25%       39.066667     18.000000     38.500000      3.666667    750.933333   
50%       42.375000     19.390000     40.900000      6.916667    756.100000   
75%       46.536000     20.600000     44.338095     10.408333    760.933333   
max       58.780000     24.500000     53.326667     26.100000    772.300000   

             RH_out     Windspeed    Visibility     Tdewpoint           rv1  \
count  19735.000000  19735.000000  19735.000000  19735.000000  19735.000000   
mean      79.750418      4.039752     38.330834      3.760707     24.988033   
std       14.901088      2.451221     11.794719      4.194648     14.496634   
min       24.000000      0.000000      1.000000     -6.600000      0.005322   
25%       70.333333      2.000000     29.000000      0.900000     12.497889   
50%       83.666667      3.666667     40.000000      3.433333     24.897653   
75%       91.666667      5.500000     40.000000      6.566667     37.583769   
max      100.000000     14.000000     66.000000     15.500000     49.996530   

                rv2  TARGET_energy  
count  19735.000000   19735.000000  
mean      24.988033     101.496833  
std       14.496634     104.380829  
min        0.005322      10.000000  
25%       12.497889      50.000000  
50%       24.897653      60.000000  
75%       37.583769     100.000000  
max       49.996530    1110.000000

The description of the dataset reveals some important information about the data. The count of the columns confirms our findings from 'df.info()'. Min, max and the mean values of the columns together can help us to obtain some extra initial information. The 'RH_5' column has a mean of 50.95 and the minimum value of 29.81 but the highest value equal to 96.32. The 75% of this column is 53.67 which is still approximately half of the max value. This may increase the possibility of this column to contain <b>outliers</b>. A similar case can be seen with the column 'T6' where the maximum value is 28.29 which is approximately 4 times the value of its mean which is 7.91. The 75% is 11.26 which is less than half of the maximum value. The minimum value is also a non-positive number which indicates both the posibility of outliers and <b>negative values</b>. Further research needs to be done to check if negative values are allowed or not. The columns RH_6,T_out and TARGET_energy show similar behaviours. RH_out, Visibility and Tdewpoint have minimum values which are very less as compared to the mean and may also contain outliers. T_out also contains some negative values.

To recapitulate, there are no columns which have any missing values.The columns <b>RH_5, T6, TH_6, RH_6, RH_out, Visibility, Tdewpoint, T_out and TARGET_energy</b> may require some extra attention while doing the exploratory data analysis. We will confirm any possible chances of outliers. Some domain knowledge will also be looked upon to verify if the negative values present in the columns:T6 and T_out are valid or invalid.

### Data Visualization